src/main/java/teammates/storage/api/AdminEmailsDb.java

```diff
@@ -94,7 +94,7 @@ public void deleteAllEmailsInTrashBin() throws BlobstoreFailureException{
         List<AdminEmailAttributes> emailsInTrashBin = getAdminEmailsInTrashBin();
         
         for (AdminEmailAttributes a : emailsInTrashBin){
-            if(a.getGroupReceiver() != null){
+            if (a.getGroupReceiver() != null && !JDOHelper.isDeleted(a)){
```

Comment:
This check can be removed, <code>getAdminEmailsInTrashBin</code> already does that check. Anyway, <code>isDeleted</code> should only be called for Entity classes, not EntityAttributes, I think